R/Record.R

```diff
@@ -94,6 +94,127 @@ Record <- R6::R6Class( # nolint object_name_linter
           )
         )
       }
+    },
+    print = function(style = TRUE) {
+      cli::cat_line(self$to_string(style))
+    },
+    to_string = function(style = FALSE) {
+      field_order <- c(
+        # Simple fields
+        "uid",
+        "handle",
+        "name",
+        "root",
+        "n",
+        "dtype",
+        "unit",
+        "description",
+        "synonyms",
+        "key",
+        "suffix",
+        "type",
+        "size",
+        "source_code",
+        "registry",
+        "hash",
+        "n_objects",
+        "n_observations",
+        "started_at",
+        "finished_at",
+        "is_consecutive",
+        "reference",
+        "reference_type",
+        "visibility",
+        "version",
+        "is_latest",
+        "region",
+        "instance_uid",
+        "created_at",
+        "updated_at",
+
+        # Relational fields
+        "created_by",
+        "storage",
+        "transform",
+        "transforms",
+        "run",
+        "report",
+        "environment",
+        "meta_artifact",
+        "ulabels",
+        "predecessors",
+        "successors",
+        "runs",
+        "parent",
+        "parents",
+        "children",
+        "output_artifacts",
+        "input_artifacts",
+        "output_collections",
+        "input_collections",
+        "input_of_runs",
+        "feature_sets",
+        "collections",
+        "features",
+        "artifacts",
+        "values",
+        "created_transforms",
+        "created_runs",
+        "created_artifacts"
+      )
+
+      column_names <- map(private$.registry$get_fields(), "column_name") |>
+        unlist()
+      column_names <- names(column_names)
+
+      # Reorder names according to set order
+      field_names <- intersect(field_order, column_names)
```

Comment:
I did this because Alex wanted the order to be deterministic and match Python. I've compromised and reduced this to a few "important" fields which get moved to the front.

Reply:
:+1: